tests/test_state_change_block.py

```diff
@@ -76,3 +76,43 @@ def test_state_name_property(self):
         blk.process_signals([StateSignal(1, 'A')])
         self.assertDictEqual(self.last_notified[DEFAULT_TERMINAL][0].to_dict(),
                              {'prev_attr': None, 'attr': 1, 'group': None})
+
+    def test_state_change_not_persist(self):
+        """Test that initial state is not persisted"""
+        blk = StateChange()
+        self.configure_block(blk, {
+                'name': 'test_block',
+                'state_expr': '{{$state}}',
+                'initial_state': '{{1}}',
+                'group_by': '{{$group}}',
+                'exclude': False
+            })
+
+        blk.start()
+
+        # get state, causing it to be saved
+        blk.get_state('A')
+
+        # assert that no values are persisted
+        self.assertTrue(len(blk.persisted_values()) == 0)
+        blk.stop()
+
+    def test_state_change_persist(self):
+        """Test that the initial state is persisted when
+            the same value as the initial state is set
+            as the state"""
+        blk = StateChange()
+        self.configure_block(blk, {
```

Comment:
Placing this assertion directly before `configure_block` should indicate which `persisted_values` will be loaded:
`self.assertEqual(blk.persisted_values(), ['_states'])`